experiment/run_experiment.py

```diff
@@ -66,12 +66,27 @@ def read_and_validate_experiment_config(config_filename: str) -> Dict:
     # it isn't like anything expensive happens before this validation is done so
     # rerunning it is cheap.
     config = yaml_utils.read(config_filename)
-    bucket_params = {'cloud_experiment_bucket', 'cloud_web_bucket'}
-    string_params = {
-        'cloud_compute_zone', 'cloud_experiment_bucket', 'cloud_web_bucket'
-    }
+    cloud_bucket_params = {'cloud_experiment_bucket', 'cloud_web_bucket'}
+    filestore_params = {'experiment_filestore', 'web_filestore'}
+    cloud_config = {'cloud_compute_zone'}
+    string_params = cloud_config.union(cloud_bucket_params).union(
+        filestore_params)
     int_params = {'trials', 'max_total_time'}
-    required_params = int_params.union(string_params)
+    required_params = int_params.union(filestore_params)
+
+    if not config.get('local_experiment', False):
+        required_params = required_params.union(cloud_config)
+
+    # Searches alternatives for file storages settings.
+    if 'experiment_filestore' not in config:
+        assert 'cloud_experiment_bucket' in config, \
```

Comment:
I purposely don't use `assert` here and instead do `valid=False` so that we can report more than one error at a time. Can you change please?

Reply:
I see. So I guess the following `todo` comment in this function actually means: (1) it wants to log errors in this function and (2) it should except early after this `read_and_validate_experiment_config` from the whole experiment view.

https://github.com/google/fuzzbench/blob/e52d53714c7bf4f371628af68ce12926252538b8/experiment/run_experiment.py#L62-L67

Thus, we choose `valid=False` here and no assertions, in order to log more errors. Please comment here if this reply hasn't got the point.

Reply:
The TODO should probably be removed because it's definitely more of a maybe. The TODO is saying that that this is complicated by the fact that tries to find as many errors as possible instead of quitting after finding the first. I'd like to keep this behavior for now. Could you please remove the TODO and do the same instead of asserting? Also could you please mention in the docstring for this fucnction that it tries to find as many errors as possible?

Thanks!

I think your understanding of what the function does based on the TODO is correct.